release/dist/unixpkgs/pkgs.go

```diff
@@ -146,7 +152,21 @@ func (t *tgzTarget) Build(b *dist.Build) ([]string, error) {
 		return nil, err
 	}
 
-	return []string{filename}, nil
+	files := []string{filename}
+
+	if t.signer != nil {
+		sig, err := t.signer.Sign(rand.Reader, h.Sum(nil), crypto.SHA512)
```

Comment:
I'm not 100% sure, but it might be the case that `Close()` causes some last bits of buffered I/O to be written out. Since we `defer` `Close`, we actually might (non-deterministically) sign an incomplete file? Does that sound possible? It might be best to explicitly, deterministically `Close`.

(I'm guessing on the basis of libc `fclose`, which is obviously not the same as `os.File`, but the basic concept might cross languages.)

Reply:
All the writers are explicitly closed above: https://github.com/tailscale/tailscale/pull/8759/files#diff-c313c5cc009cf463fc06aae4a619fbd78ec928df4bd80e7917145610ecee694eL139-L147
The deferred `Close` calls are just fail-safe noops in this case.